pkg/services/sqlstore/migrator/migrator.go

```diff
@@ -232,49 +247,62 @@ func (mg *Migrator) run() (err error) {
 		}
 	}
 
-	mg.Logger.Info("migrations completed", "performed", migrationsPerformed, "skipped", migrationsSkipped, "duration", time.Since(start))
+	logger.Info("DB migrations completed", "performed", migrationsPerformed, "skipped", migrationsSkipped, "duration", time.Since(start))
 
 	// Make sure migrations are synced
 	return mg.DBEngine.Sync2()
 }
 
-func (mg *Migrator) exec(m Migration, sess *xorm.Session) error {
-	mg.Logger.Info("Executing migration", "id", m.Id())
+func (mg *Migrator) exec(parentCtx context.Context, m Migration, sess *xorm.Session) error {
+	start := time.Now()
+
+	ctx, span := mg.tracer.Start(parentCtx, "Migrator.exec")
+	span.SetAttributes("migration_id", m.Id(), attribute.String("migration_id", m.Id()))
+	defer span.End()
+
+	sess.Context(ctx)
+
+	logger := mg.Logger.New("id", m.Id()).FromContext(ctx)
+	logger.Info("Executing migration")
 
 	condition := m.GetCondition()
 	if condition != nil {
 		sql, args := condition.SQL(mg.Dialect)
 
 		if sql != "" {
-			mg.Logger.Debug("Executing migration condition SQL", "id", m.Id(), "sql", sql, "args", args)
+			logger.Debug("Executing migration condition SQL", "sql", sql, "args", args)
 			results, err := sess.SQL(sql, args...).Query()
 			if err != nil {
-				mg.Logger.Error("Executing migration condition failed", "id", m.Id(), "error", err)
+				logger.Error("Executing migration condition failed", "error", err)
 				return err
 			}
 
 			if !condition.IsFulfilled(results) {
-				mg.Logger.Warn("Skipping migration: Already executed, but not recorded in migration log", "id", m.Id())
+				logger.Warn("Skipping migration: Already executed, but not recorded in migration log")
 				return nil
 			}
 		}
 	}
 
 	var err error
 	if codeMigration, ok := m.(CodeMigration); ok {
-		mg.Logger.Debug("Executing code migration", "id", m.Id())
+		logger.Debug("Executing code migration")
 		err = codeMigration.Exec(sess, mg)
 	} else {
 		sql := m.SQL(mg.Dialect)
-		mg.Logger.Debug("Executing sql migration", "id", m.Id(), "sql", sql)
+		logger.Debug("Executing sql migration", "sql", sql)
 		_, err = sess.Exec(sql)
 	}
 
 	if err != nil {
-		mg.Logger.Error("Executing migration failed", "id", m.Id(), "error", err)
+		logger.Error("Migration failed", "error", err)
+		span.SetStatus(codes.Error, err.Error())
+		span.RecordError(err)
 		return err
 	}
 
+	logger.Info("Migration successfully executed", "duration", time.Since(start))
```

Comment:
Maybe this would be enough of an improvement for now, e.g. logging the duration of each migration?

Reply:
> Maybe this would be enough of an improvement for now, e.g. logging the duration of each migration, and skipping distributed tracing? 

I've never really felt the need for tracing migrations so logs might be enough.

Reply:
Sounds good :+1: